core/templates/dev/head/services/assets-backend-api.service.spec.ts

```diff
@@ -108,20 +147,158 @@ describe('Assets Backend API Service', function() {
+  it('should handler rejection on fetching file when Blob throws a' +
+    ' TypeErrow and BlobBuilder does not exist', function() {
+    var successHandler = jasmine.createSpy('success');
+    var failHandler = jasmine.createSpy('fail');
 
+    spyOn(window, 'Blob').and.callFake(function() {
+      throw new TypeError();
+    });
+
+    $httpBackend.expect('GET', audioRequestUrl).respond(201, 'Error');
+    expect(AssetsBackendApiService.isCached('myfile.mp3')).toBe(false);
+
+    AssetsBackendApiService.loadAudio('0', 'myfile.mp3').then(
+      successHandler, failHandler);
+    $httpBackend.flush();
+    expect(successHandler).not.toHaveBeenCalled();
+    expect(failHandler).toHaveBeenCalledWith('myfile.mp3');
+    $httpBackend.verifyNoOutstandingExpectation();
+  });
+
+  it('should handler rejection on fetching file when Blob throws a' +
+    ' TypeError and BlobBuilder does not work correctly', function() {
+    var successHandler = jasmine.createSpy('success');
+    var failHandler = jasmine.createSpy('fail');
+
+    spyOn(window, 'Blob').and.callFake(function() {
+      throw new TypeError();
+    });
+
+    window.BlobBuilder = function BlobBuilder() {
+      this.blob = new Object({ blobBuilder: true });
+    };
+
+    $httpBackend.expect('GET', audioRequestUrl).respond(201, 'Error');
+    expect(AssetsBackendApiService.isCached('myfile.mp3')).toBe(false);
+
+    AssetsBackendApiService.loadAudio('0', 'myfile.mp3').then(
+      successHandler, failHandler);
+    $httpBackend.flush();
+    expect(successHandler).not.toHaveBeenCalled();
+    expect(failHandler).toHaveBeenCalledWith('myfile.mp3');
+    $httpBackend.verifyNoOutstandingExpectation();
+  });
+
+  it('should fetch file when Blob throws a TypeError and BlobBuilder' +
+    ' is correctly implemented', function() {
+    var successHandler = jasmine.createSpy('success');
+    var failHandler = jasmine.createSpy('fail');
+
+    spyOn(window, 'Blob').and.callFake(function() {
+      throw new TypeError();
+    });
+
+    window.BlobBuilder = function BlobBuilder() {
+      this.blob = new Object({ blobBuilder: true });
+      this.append = function append(data) {};
+      this.getBlob = function getBlob(contentType) {
+        return this.blob;
+      };
+    };
+
+    $httpBackend.expect('GET', audioRequestUrl).respond(201, 'Error');
+    expect(AssetsBackendApiService.isCached('myfile.mp3')).toBe(false);
+
+    AssetsBackendApiService.loadAudio('0', 'myfile.mp3').then(
+      successHandler, failHandler);
+    $httpBackend.flush();
+    expect(successHandler).toHaveBeenCalledWith(
+      audioFileObjectFactory.createNew(
+        'myfile.mp3',
+        { blobBuilder: true }
+      ));
+    expect(failHandler).not.toHaveBeenCalled();
+    $httpBackend.verifyNoOutstandingExpectation();
+  });
+
+  it('should successfully save an audio', function(done) {
+    var successMessage = 'Audio was successfully saved.';
+    // @ts-ignore
```

Comment:
Why this pragma? (Can we add the type?) [Please add a comment if it's not possible to add the type here.]

Reply:
Done.